File: core/feature04a_test.js
package main

import (
	"testing"
)

var (
	feature04ANames      = []string{"M"}
	feature04AAccuracies = []string{"%.0f"}
	feature04ATypes      = []bool{true}
)

func Test01Feature04A(t *testing.T) {
	arguments := map[string]interface{}{"A": 40}
	expected := map[string]interface{}{"M": 773}
	genericTester(t, feature04A(arguments), expected, feature04ANames, feature04AAccuracies, feature04ATypes)
}

File: core/feature04a_test.go
package main

import (
	"testing"
)

func feature04ATester(t *testing.T, aa float64, mmOK float64) {
	mm := feature04A(aa)
	if !equalsFormatted(mm, mmOK, "%.0f") {
		t.Errorf("The value 'mm' is incorrect (got: %v, expected: %v)!", mm, mmOK)
	}
}

func Test01Feature04A(t *testing.T) {
	feature04ATester(t, 40, 773)
}
